arangod/Cluster/TraverserEngine.cpp

```diff
@@ -122,14 +122,16 @@ BaseEngine::BaseEngine(TRI_vocbase_t& vocbase, aql::QueryContext& query,
   for (auto collection : VPackObjectIterator(vertexSlice)) {
     ResourceUsageAllocator<MonitoredCollectionToShardMap, ResourceMonitor>
         alloc = {_query.resourceMonitor()};
-    MonitoredStringVector shards{alloc};
+    MonitoredShardIDVector shards{alloc};
     TRI_ASSERT(collection.value.isArray());
     for (auto shard : VPackArrayIterator(collection.value)) {
       TRI_ASSERT(shard.isString());
-      std::string name = shard.copyString();
-      _query.collections().add(name, AccessMode::Type::READ,
+      auto shardString = shard.copyString();
```

Comment:
```suggestion
      auto shardString = shard.stringView();
```

Reply:
Collections.add requires the std::string.
Would like to avoid changing it's signature with this PR as well.
Therefore the copy has to happen anyways.